src/main/java/Duke.java

```diff
@@ -1,10 +1,258 @@
+        if(splitDeadline.length != 2) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_WRONG_FORMAT);
+            System.out.println(MESSAGE_CORRECT_DEADLINE_FORMAT);
+            System.out.println(OUTPUT_DIVIDER);
+            return wasAdded;
+        }
+
+        String[] deadlineDescriptionAndDeadline = splitDeadline[1].split(DEADLINE_PREFIX, 2);
+        if(deadlineDescriptionAndDeadline.length != 2) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_WRONG_FORMAT);
+            System.out.println(MESSAGE_CORRECT_DEADLINE_FORMAT);
+            System.out.println(OUTPUT_DIVIDER);
+            return wasAdded;
+        }
+
+        String deadlineDescription = deadlineDescriptionAndDeadline[0].trim();
+        String deadlineDeadline = deadlineDescriptionAndDeadline[1].trim();
+        boolean isValidDeadline = !deadlineDescription.isEmpty() && !deadlineDeadline.isEmpty();
+        // deadline with valid format is added to task list
+        if (isValidDeadline) {
+            Deadline newDeadline = new Deadline(deadlineDescription, deadlineDeadline);
+            tasks[taskCount] = newDeadline;
+            wasAdded = true;
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_TASK_ADDED);
+            System.out.println("   " + newDeadline.toString());
+            System.out.println(" You now have " + (taskCount + 1) + " task(s) in the list.");
+            System.out.println(OUTPUT_DIVIDER);
+        } else {
+            wasAdded = false;
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_LACKS_ARG);
+            System.out.println(MESSAGE_CORRECT_DEADLINE_FORMAT);
+            System.out.println(OUTPUT_DIVIDER);
+        }
+        return wasAdded;
+    }
+
+    public static boolean canAddAndPrintEvent(Task[] tasks, int taskCount, String addedEvent) {
+        String[] splitEvent = addedEvent.split("\\s+", 2);
+        boolean wasAdded = false;
+
+        // Makes sure that the event entered is of the format "event {description} /at {event time}"
+        if(splitEvent.length != 2) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_WRONG_FORMAT);
+            System.out.println(MESSAGE_CORRECT_EVENT_FORMAT);
+            System.out.println(OUTPUT_DIVIDER);
+            return wasAdded;
+        }
+
+        String[] eventDescriptionAndTime = splitEvent[1].split(EVENT_PREFIX, 2);
+        if (eventDescriptionAndTime.length != 2) {
+            System.out.println(OUTPUT_DIVIDER);
+            System.out.println(MESSAGE_COMMAND_WRONG_FORMAT);
+            System.out.println(MESSAGE_CORRECT_EVENT_FORMAT);
+            System.out.println(OUTPUT_DIVIDER);
+            return wasAdded;
+        }
+
+        String eventDescription = eventDescriptionAndTime[0].trim();
```

Comment:
This line of code and the one below seems abit out of place, should it be there or can it be logically placed into a another method?